test/unordered/constructor_tests.cpp

```diff
@@ -317,66 +329,208 @@ namespace constructor_tests {
     }
 
 #if !defined(BOOST_NO_CXX11_HDR_INITIALIZER_LIST)
-    std::initializer_list<typename T::value_type> list;
+    typedef typename T::value_type value_type;
+
+    std::initializer_list<value_type> list;
+
+    test::random_values<T> v(3, generator);
+    std::vector<value_type> vec(v.begin(), v.end());
+    BOOST_ASSERT(vec.size() >= 3);
+
+    std::initializer_list<value_type> nonempty_list{vec[0], vec[1], vec[2]};
```

Comment:
This looks fishy to me. The backing array of the initializer_list is destroyed at the end of the full expression.

Reply:
Hmm, that's a good question.

It seems to pass both sanitizers and checks for the values. For example, we still pass the check where we compare the container's contents to `expected`.

Reply:
Yeah, declaring a variable of type initializer_list does extend the lifetime of the temporary backing array, https://eel.is/c++draft/dcl.init.list#6. So it's fine.